libs/parallelism/algorithms/include/hpx/parallel/algorithms/is_sorted.hpp

```diff
@@ -34,25 +249,28 @@ namespace hpx { namespace parallel { inline namespace v1 {
     // is_sorted
     namespace detail {
         /// \cond NOINTERNAL
-        template <typename FwdIter>
-        struct is_sorted : public detail::algorithm<is_sorted<FwdIter>, bool>
+        template <typename FwdIter, typename Sent>
+        struct is_sorted
+          : public detail::algorithm<is_sorted<FwdIter, Sent>, bool>
         {
             is_sorted()
               : is_sorted::algorithm("is_sorted")
             {
             }
 
-            template <typename ExPolicy, typename Pred>
+            template <typename ExPolicy, typename Pred, typename Proj>
             static bool sequential(
-                ExPolicy, FwdIter first, FwdIter last, Pred&& pred)
+                ExPolicy, FwdIter first, Sent last, Pred&& pred, Proj&& proj)
             {
-                return std::is_sorted(first, last, std::forward<Pred>(pred));
+                return is_sorted_sequential(first, last,
+                    util::invoke_projected<Pred, Proj>(
+                        std::forward<Pred>(pred), std::forward<Proj>(proj)));
```

Comment:
Now, that you have your own implementations of `is_sorted`, may I suggest to directly pass the projection to that implementation instead of having to create another indirection layer (the `invoke_projected`)?

Reply:
Also a good idea! I've changed that as well.